src/Controller.cpp

```diff
@@ -117,11 +117,20 @@ bool Controller::isRTR(CANFrame *amsg)
 
 void Controller::indicateMode(byte mode)
 {
-  newMode = mode;  // used by Event Producer Service
   // DEBUG_SERIAL << F("> indicating mode = ") << mode << endl;
   if (_ui) {
     _ui->indicateMode(mode);
   }
+  
+  if (mode == MODE_NORMAL) // used by Event Producer Service
```

Comment:
I wonder if this is a bit blunt. Isn't this condition true for any transition to NORMAL?

Reply:
The route by which it is called only occurs if there has been a transition from MODE_SETUP to MODE_NORMAL.  coming to or from LEARN or NHEARTBEAT does not cause a change to indicate mode.  Thus, it is only if we have requested a node number or returned to the old one on timeout that it is called, which is just what we want.